python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1480,6 +1491,73 @@ def latest_materialization_event(
             AssetKey.from_coercible(key)
         )
 
+    ##### Partition methods
+
+    @public
+    @property
+    def has_partition_key(self) -> bool:  # TODO - maybe this one can be replaced by something else?
+        return self.op_execution_context.has_partition_key
+
+    @public
+    @property
+    def partition_key(self) -> str:
+        return self.op_execution_context.partition_key
+
+    @public
+    @property
+    def partition_keys(self) -> Sequence[str]:
+        return self.op_execution_context.partition_keys
+
+    @public
+    @property
+    def partition_key_range(self) -> PartitionKeyRange:
+        return self.op_execution_context.partition_key_range
+
+    @public
+    @property
+    def partition_time_window(self) -> TimeWindow:
+        return self.op_execution_context.partition_time_window
+
+    @public
+    def upstream_partition_key(self, key: CoercibleToAssetKey) -> str:
+        return self._step_execution_context.asset_partition_key_for_upstream(
+            AssetKey.from_coercible(key)
+        )
+
+    @public
+    def upstream_partition_keys(self, key: CoercibleToAssetKey) -> Sequence[str]:
```

Comment:
make all of these accept a `CoercibleToAssetDep` maybe? that way we can pass definitions around vs string matching on asset key? or maybe doing this is sufficiently good pattern 

```python
@asset(...)
def upstream():
   ...

@asset(deps=[upstream], ...)
def downstream(context):
      context.upstream_partition_keys(upstream.key)
```

allowing `CoercibleToAssetDep` could cause confusing scoping issues like this 

```python
@asset(...)
def upstream():
   ...

@asset(...)
def downstream(context, upstream):
      context.upstream_partition_keys(upstream.key) # upstream doesn't have a key in this case since it's a loaded value
```